### PR TITLE
Parent Readme Update

### PR DESCRIPTION
**Pull request by sourav-kundu**
Suggest changes if any

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,64 @@
 # BrowserStack GitHub Actions
 
-This respository contains the GitHub Actions to help you integrate your test suite with BrowserStack cloud.
+This respository contains a library of GitHub Actions to help you integrate your test suite with the [BrowserStack](https://browserstack.com) device cloud. 
+
+You need a BrowserStack username and access-key to run your tests on the BrowserStack device cloud. You can [sign-up for free trial](https://www.browserstack.com/users/sign_up) if you do not have an existing account. We love open source projects. If you want to test your open source project on BrowserStack then [sign-up](https://www.browserstack.com/open-source) for your lifetime free access to all our products.
```
Comment (rchougule) [inline README.md]:
```suggestion
You need a BrowserStack username and access-key to run your tests on the BrowserStack device cloud. You can [sign-up for free trial](https://www.browserstack.com/users/sign_up) if you do not have an existing account. 
If you want to test your open source project on BrowserStack then [sign-up here](https://www.browserstack.com/open-source) for lifetime free access to all our products.
```
How about trying to separate out the `open-source` part and focusing on it on a new line ? ...

File: README.md [hunk 2]
```diff
@@ -1,11 +1,64 @@
...
 
 ## Available Actions
-* [setup-env](./setup-env): This action helps in setting up of environment variables to be used in your test scripts for BrowserStack. The environment variables set up here shall be used by other BrowserStack actions as well for their functioning.
+* [setup-env](./setup-env): This Action helps in setting up the required environment variables that are to be used in your test scripts. The environment variables set up here shall be used by other BrowserStack actions as well for their functioning.
   
-* [setup-local](./setup-local): This actions sets up local binary for creating local tunnel connections from runner environment to BrowserStack cloud.
+* [setup-local](./setup-local): This actions sets up downloads and starts the appropriate BrowserStackLocal binary thereby creating a secure tunnel connection from the GitHub Actions runner environment to the BrowserStack device cloud. This secure tunnel will be used by the remote browsers in BrowserStack to access your web application hosted in the GitHub Actions runner machine. You do not need this Action, if the application to be tested is accessible over public internet.
```
Comment (rchougule) [inline README.md]:
```suggestion
* [setup-local](./setup-local): This Action downloads and starts the appropriate BrowserStackLocal binary, thereby creating a secure tunnel connection from the GitHub Actions runner environment to the BrowserStack device cloud. This secure tunnel will be used by the remote browsers in BrowserStack to access your web application hosted in the GitHub Actions runner machine. You do not need this Action, if the application to be tested is accessible over public internet.
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,64 @@
...
+
+## Pre-requisites
+* You should set your BrowserStack Username and Access-key as GitHub Secrets `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` respectively.
```
Comment (rchougule) [inline README.md]:
```suggestion
* You should set your BrowserStack Username and Access Key as GitHub Secrets `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` respectively.
```

File: README.md [hunk 4]
```diff
@@ -1,11 +1,64 @@
...
+
+## Usage
+As this is actually a library of Actions, invoking only this Action will only invoke the `setup-env` Action internally. The following usage example will only set up the required environment variables:
```
Comment (rchougule) [inline README.md]:
```suggestion
As this is a library of Actions, invoking this Action will trigger the `setup-env` Action internally. The following usage example will **only** set up the required environment variables:
```
minor refactor...

File: README.md [hunk 5]
```diff
@@ -1,11 +1,65 @@
...
 
 ## Available Actions
-* [setup-env](./setup-env): This action helps in setting up of environment variables to be used in your test scripts for BrowserStack. The environment variables set up here shall be used by other BrowserStack actions as well for their functioning.
+* [setup-env](./setup-env): This Action helps in setting up the required environment variables that are to be used in your test scripts. The environment variables set up here shall be used by other BrowserStack actions as well for their functioning.
   
-* [setup-local](./setup-local): This actions sets up local binary for creating local tunnel connections from runner environment to BrowserStack cloud.
+* [setup-local](./setup-local): This Action downloads and starts the appropriate BrowserStackLocal binary, thereby creating a secure tunnel connection from the GitHub Actions runner environment to the BrowserStack device cloud. This secure tunnel will be used by the remote browsers in BrowserStack to access your web application hosted in the GitHub Actions runner machine. You do not need this Action, if the application to be tested is accessible over public internet.
```
Comment (rchougule) [inline README.md]:
```suggestion
* [setup-local](./setup-local): This Action downloads and starts the appropriate BrowserStackLocal binary, thereby creating a secure tunnel connection from the GitHub Actions runner environment to the BrowserStack device cloud. This secure tunnel will be used by the remote browsers in BrowserStack to access your web application hosted in the GitHub Actions runner environment. **You do not need this Action if the application to be tested is accessible over the public internet.**
```

File: README.md [hunk 6]
```diff
@@ -1,11 +1,65 @@
...
+
+## Pre-requisites
+* You should set your BrowserStack Username and Access Key as GitHub Secrets `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` respectively.
```
Comment (rchougule) [inline README.md]:
```suggestion
* You should set your BrowserStack Username and Access Key as GitHub Secrets, i.e. `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` respectively.
```

File: README.md [hunk 7]
```diff
@@ -1,11 +1,65 @@
...
+  with:
+    username:  ${{ secrets.BROWSERSTACK_USERNAME }}
+    access-key: ${{ secrets.BROWSERSTACK_ACCESS_KEY }}
+    build-name: BUILD_INFO
+    project-name: REPO_NAME
+```
+We recommend you to invoke individual Actions from this library separately. A sample of two Actions being invoked back to back is shown below. You can additionally refer to the individual `README` ([setup-env](./setup-env), [setup-local](./setup-local)) of the Actions to know more about how they work, the inputs they support and their usage examples.
+
+## Sample Workflow with usage of both Actions
+The workflow example below would be useful for anyone who wants to run their automated tests on the BrowserStack cloud when the web application to be tested, is hosted on a GitHub Actions runner i.e. not accessible from public Internet
```
Comment (rchougule) [inline README.md]:
```suggestion
The workflow example below would be useful when the web application to be tested is hosted on the GitHub Actions runner environment, i.e. not accessible from the public Internet.
```

File: README.md [hunk 8]
```diff
@@ -1,11 +1,65 @@
...
+          access-key: ${{ secrets.BROWSERSTACK_ACCESS_KEY }}
+          build-name: BUILD_INFO
+          project-name: REPO_NAME
+      - name: 'BrowserStackLocal Setup'
+        uses: 'browserstack/github-actions/setup-local@master'
+        with:
+          local-testing: start
+          local-identifier: random
+```
+Post these setup steps where both the Actions have been used, you have to build and run your application web server on the same machine, in daemon mode and thereafter invoke your test scripts. Your test scripts should use the environment variables that have been set by [`setup-env` Action](./setup-env). For more detailed steps on how to integrate your test suite with BrowserStack on GitHub Actions, visit [BrowserStack Documentation](http://browserstack.com/docs/automate/selenium/github-actions) for the same.
```
Comment (rchougule) [inline README.md:55]: we can modify this as bullet points and put it in a `Note` section...

**Review by rchougule — CHANGES_REQUESTED** [review]
minor changes requested

File: README.md (hunk 8 above)
Comment (sourav-kundu) [reply README.md:55]: Merge all of it with your branch...
then make necessary changes as you deem fit...
Finally show me once after all changes have been done

File: README.md [hunk 9]
```diff
@@ -1,11 +1,65 @@
 # BrowserStack GitHub Actions
 
-This respository contains the GitHub Actions to help you integrate your test suite with BrowserStack cloud.
+This respository contains a library of GitHub Actions to help you integrate your test suite with the [BrowserStack](https://browserstack.com) device cloud. 
+
+You need a BrowserStack username and access-key to run your tests on the BrowserStack device cloud. You can [sign-up for free trial](https://www.browserstack.com/users/sign_up) if you do not have an existing account. 
+If you want to test your open source project on BrowserStack then [sign-up here](https://www.browserstack.com/open-source) for lifetime free access to all our products.
```
Comment (rchougule) [inline README.md:6]:
```suggestion
You need a BrowserStack username and access-key to run your tests on the BrowserStack device cloud. You can [sign-up for free trial](https://www.browserstack.com/users/sign_up) if you do not have an existing account.

If you want to test your open source project on BrowserStack then [sign-up here](https://www.browserstack.com/open-source) for lifetime free access to all our products.
```

File: README.md (hunk 8 above)
Comment (rchougule) [reply README.md:55]: will do

**Pull request merged**